torch/_inductor/graph.py

```diff
@@ -578,16 +599,17 @@ def mark_buffer_mutated(self, name: str):
 
     def add_tensor_constant(self, data, name=None):
         def allocate(name):
-            for constant_name, value in self.constants.items():
-                if (
-                    not data.is_mkldnn
-                    and data.size() == value.size()
-                    and data.stride() == value.stride()
-                    and data.dtype == value.dtype
-                    and data.device == value.device
-                    and torch.eq(data, value).all()
-                ):
-                    return constant_name
+            if not config.aot_inductor.use_runtime_constant_folding:
```

Comment:
Why dont we use this instead: https://github.com/pytorch/pytorch/blob/main/torch/_inductor/config.py#L282-L284

Reply:
Ah.. I think it's slightly different. `always_keep_tensor_constants` prevents us inlining constants directly, however it allows reusing of the constants (this if statement checks if the to-be-added tensor has a value exactly the same as existing constants.)
We add the if here mainly because we want to make sure that all constants get generated "even if we have multiple constants with same value". Since we can't guarantee that constants wouldn't be updated.